packages/bierzo-wallet/src/store/rpcendpoint/selectors.ts

```diff
@@ -0,0 +1,7 @@
+import { RpcEndpointType } from "../../communication/rpcEndpoint";
+import { RootState } from "../reducers";
+
+export const getRpcEndpointType = (state: RootState): RpcEndpointType | undefined => {
+  const rpcEndpointType = state.rpcEndpoint;
+  return rpcEndpointType ? rpcEndpointType.rpcEndpointType : undefined;
```

Comment:
What about throwing an exception if `state.rpcEndpoint` is unset and return type `RpcEndpointType`?

Reply:
Hmm.. yes we can. Because we are 100% sure that this state should be set before we reach the balance route.

Reply:
And in this case we will not need to make `readonly rpcEndpointType: RpcEndpointType | undefined;` optional

Reply:
> Because we are 100% sure that this state should be set before we reach the balance route.

This is at least the case for using the app. Not sure about tests